python/pylibcudf/pylibcudf/io/types.pyx

```diff
@@ -288,12 +288,14 @@ cdef class TableInputMetadata:
     """
     def __init__(self, Table table):
         self.c_obj = table_input_metadata(table.view())
-        self.column_metadata = [
+
+    @property
+    def column_metadata(self):
+        return [
             ColumnInMetadata.from_libcudf(&self.c_obj.column_metadata[i], self)
             for i in range(self.c_obj.column_metadata.size())
         ]
 
```

Comment:
*question*: What is this change needed for?

Reply:
Before this was an (unused) `cdef` attribute, but cudf Python ORC handling modifies the `column_metadata` inplace, so I needed to expose this attribute in Python space.

Reply:
Can you add it as a property in types.pyi if it is not already done?

Reply:
Sure. Added in [5e27f2a](https://github.com/rapidsai/cudf/pull/17466/commits/5e27f2ae07574b02a2006baafbe79c455e9a26f0)